pallets/file-system/src/utils.rs

```diff
@@ -2063,6 +2068,13 @@ where
                 // Decrease size of the bucket.
                 <T::Providers as MutateBucketsInterface>::decrease_bucket_size(&bucket_id, size)?;
 
+                // If the bucket has a MSP, decrease its used capacity.
+                if let Some(msp_id) = msp_id {
+                    <T::Providers as MutateStorageProvidersInterface>::decrease_capacity_used(
+                        &msp_id, size,
+                    )?;
+                }
```

Comment:
can you add the checks in the unit tests for this? as well for the other place you added it in